src/request.go

```diff
@@ -102,7 +102,19 @@ func insertDomainMetrics(eventType string, domain string, beanAttrVals []*beanAt
 	// Create an entity for the domain
 	var e *integration.Entity
 	var err error
-	if args.LocalEntity {
+	if args.RemoteMonitoring {
+		if args.ConnectionURL != "" {
+			e, err = newRemoteEntity(domain, getConnectionUrlSAP(args.ConnectionURL), i)
+			if err != nil {
+				return err
+			}
+		} else {
+			e, err = newRemoteEntity(domain, fmt.Sprintf("%s:%s", host, port), i)
+			if err != nil {
+				return err
+			}
+		}
+	} else if args.LocalEntity {
```

Comment:
not a big issue but I would simplify this to:

```go
	if args.RemoteMonitoring {
		url := fmt.Sprintf("%s:%s", host, port)
		if args.ConnectionURL != "" {
			url = getConnectionUrlSAP(args.ConnectionURL)
		}
		e, err = newRemoteEntity(domain, url, i)
		if err != nil {
			return err
		}
	}
```

Reply:
thank you, this makes code simpler.